### PR TITLE
The user who created a recipe can edit it

### PR DESCRIPTION
**Pull request by tochman**
PT: https://www.pivotaltracker.com/story/show/169736108

```
As a User
After I have created my recipe
I would like to be able to edit it
```

### Notes
Response for unauthorized request:
```json
{"errors":["You need to sign in or sign up before continuing."]}
```

### CODE REVIEW

File: spec/requests/v1/recipes_spec.rb [hunk 1]
```diff
@@ -3,11 +3,47 @@
...
+  let(:HTTP_CLIENT) { user_credentials['client'] }
+  let(:HTTP_EXPIRY) { user_credentials['expiry'] }
+  let(:HTTP_UID) { user_credentials['uid'] }
+
   path '/v1/recipes' do
     post 'Creates a recipe' do
       tags 'Recipes'
       description 'Creates an instance of Recipe.'
       consumes 'application/json'
+      parameter name: :HTTP_ACCESS_TOKEN,
```
Comment (oliverochman) [inline spec/requests/v1/recipes_spec.rb:18]: is there a way of extracting these parameters, clutters up the test

File: spec/requests/v1/recipes_spec.rb [hunk 2]
```diff
@@ -89,5 +127,60 @@
         run_test!
       end
     end
+
+    put 'Edits a recipe' do
+      tags 'Recipes'
```
Comment (oliverochman) [inline spec/requests/v1/recipes_spec.rb:132]: Why did you choose to only test this with swagger?

File: app/controllers/v1/recipes_controller.rb [hunk 3]
```diff
@@ -23,6 +25,11 @@ def show
     render json: recipe, serializer: Recipes::ShowSerializer
   end
 
+  def update
+    recipe = Recipe.find(params[:id])
+    render json: recipe, serializer: Recipes::ShowSerializer, status: 201
```
Comment (oliverochman) [inline app/controllers/v1/recipes_controller.rb]: Why are you sending the recipe back? Should we not send a success message instead?

File: spec/requests/v1/recipes_spec.rb [hunk 4]
```diff
@@ -3,11 +3,47 @@
...
+
+      parameter name: :HTTP_UID,
+                in: :header,
+                type: :string,
+                example: 'user@mail.com',
+                required: true
+
       parameter name: :recipe, in: :body, schema: {
         type: :object,
         properties: {
```
Comment (clalil) [inline spec/requests/v1/recipes_spec.rb:49]: I dont see a title here, isn't that mandatory?

File: app/controllers/v1/recipes_controller.rb (hunk 3 above)
Comment (oliverochman) [reply app/controllers/v1/recipes_controller.rb]: what happens if we try to update the ingredients of a recipe with more then 500 characters?

`validates_length_of :ingredients, maximum: 500`

File: spec/requests/v1/recipes_spec.rb (hunk 4 above)
Comment (clalil) [reply spec/requests/v1/recipes_spec.rb:49]: Of fudge, we never had the validation for it to have a title, right?

**Review by yasmineezequiel — APPROVED** [review]

File: spec/requests/v1/recipes_spec.rb (hunk 4 above)
Comment (tochman) [reply spec/requests/v1/recipes_spec.rb:49]: We do, but in this case I only need an ID I make the call

File: spec/requests/v1/recipes_spec.rb (hunk 1 above)
Comment (tochman) [reply spec/requests/v1/recipes_spec.rb:18]: Not at this point

File: spec/requests/v1/recipes_spec.rb (hunk 2 above)
Comment (tochman) [reply spec/requests/v1/recipes_spec.rb:132]: I'm waiting for the other PR to be merged

File: app/controllers/v1/recipes_controller.rb (hunk 3 above)
Comment (tochman) [reply app/controllers/v1/recipes_controller.rb]: question 1: The Recipe has new attributes so I thought we might want to get it back. I'm sending a success code with the response.

Comment (tochman) [reply app/controllers/v1/recipes_controller.rb]: question 2: we are NOT testing for this now directly. But the current shoulda_matcher test does it for us.

File: app/controllers/v1/recipes_controller.rb [hunk 5]
```diff
@@ -26,9 +28,9 @@ def show
   def update
     recipe = Recipe.find(params[:id])
     if recipe.update(recipe_params)
-      render json: { message: 'Your recipe has been updated.' }, status: 200
+      render json: { message: 'Your recipe has been updated.' }, status: 201
```
Comment (oliverochman) [inline app/controllers/v1/recipes_controller.rb:31]: please change this to a 200 response

File: config/initializers/devise_token_auth.rb [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+  # this to false to prevent the Authorization header from changing after
+  # each request.
+  # config.change_headers_on_each_request = true
+
+  # By default, users will need to re-authenticate after 2 weeks. This setting
+  # determines how long tokens will remain valid after they are issued.
+  # config.token_lifespan = 2.weeks
+
+  # Limiting the token_cost to just 4 in testing will increase the performance of
+  # your test suite dramatically. The possible cost value is within range from 4
```
Comment (oliverochman) [inline config/initializers/devise_token_auth.rb:15]: I need you to set some settings in here that we have in all CA projects

```
  config.change_headers_on_each_request = false
  config.token_lifespan = 2.weeks
  config.batch_request_buffer_throttle = 5.seconds
```

**Review by yasmineezequiel — APPROVED** [review]

**Review by oliverochman — APPROVED** [review]

**Review by shsteindorsson — APPROVED** [review]

File: app/controllers/v1/recipes_controller.rb (hunk 5 above)
Comment (tochman) [reply app/controllers/v1/recipes_controller.rb:31]: Yeah, I agree. on second thought, it should be 200

File: config/initializers/devise_token_auth.rb (hunk 6 above)
Comment (tochman) [reply config/initializers/devise_token_auth.rb:15]: okay, will do

Comment (tochman) [reply config/initializers/devise_token_auth.rb:15]: ..in another PR ;-)

**Pull request merged**